### DISCUSSION TITLE
How to send POST request with body using http.fetch

### DISCUSSION THREAD

**Discussion by lumiaczek** [tauri-apps/tauri#4955] [Q&A] [answered]
Hello, when i try to send a fetch POST request to my Express server it throws an Error:
`Unhandled Promise Rejection: invalid type: string "{\"login\":\"\",\"password\":\"\"}", expected adjacently tagged enum Body`

I am using SvelteKit and this is my script tag section:
```html 
<script context="module">
  import { goto } from "$app/navigation";
  import { fetch } from "@tauri-apps/api/http";
</script>

<script>
  let user = {
    login: "",
    password: "",
  };

  const handleLogin = () => {
    fetch("http://localhost:5000/login", {
      method: "POST",
      body: JSON.stringify(user),
    });
  };
</script>
```
I think that `JSON.stringify()` is causing an error but i have no idea how to solve this.

**Comment by FabianLars** [comment, marked as answer]
`body` needs to be an instance of the Body class: https://tauri.app/v1/api/js/classes/http.Body 

I guess we should look into manually checking the input type before sending it to rust to get better errors here.

**Comment by sqmw** [comment]
you can use http.Body.json({}) and so on，you use now maybe a bug
import {http} from '@tauri-apps/api';
